lib/pagseguro/collection_base.rb

```diff
@@ -0,0 +1,41 @@
+module PagSeguro
+  class CollectionBase
+    extend CollectionManagers::MacroMethods
+    extend Forwardable
+    include Enumerable
+
+    def_delegators :@store, :size, :clear, :empty?, :any?, :each
+
+    def initialize
+      @store = []
+    end
+
+    def collection_manager
+      CollectionManagers::GenericManager
+    end
+
+    def collection_type
+      eval singularize(self.class.to_s)
```

Comment:
yes it has, but singularize returns me `'PagSeguro::Class'` as *String*, so for me use `#const_get` I need to pass just one arg like `'Class'` this way I'll have to split *::* and take only the class to use. I'll see how I can improve this. :+1: